base_url/models/model.py

```diff
@@ -0,0 +1,143 @@
+        return url_key
+
+    @api.multi
+    def _inverse_set_url(self):
+        """
+        backup old url
+
+        1 find url redirect true and same model_id
+        if other model id refuse
+        2 if exists set to False
+
+        3 write the new one
+        """
+
+        model_ref = "%s,%s" % (self._name, self.id)
+
+        # key exist ?
+        already_exist_url = self.env['url.url'].search(
+            [('url_key', '=', self.url_key)])
+
+        if already_exist_url:
+            # existing key in wich object ?
+            for model in already_exist_url.model_id:
+                model_txt = "%s,%s" % (model._name, model.id)
+                if model_txt != model_ref:
+                    # existing key for other model
+                    raise UserError(
+                        _("Url_key already exist in other model"
+                          " %s" % (model.name)))
+                else:  # existing key for same object toggle redirect to False
+                    already_exist_url.redirect = False
+        else:
+            # no existing key creating one
+            vals = {'url_key': self.url_key,
+                    'model_id': model_ref,
+                    'redirect': False}
+            self.env['url.url'].create(vals)
+            # other url of object set redirect to True
+        this_model_urls = self.env['url.url'].search([
+            ('model_id', '=', model_ref),
+            ('url_key', '!=', self.url_key)])
+
+        for exist_url in this_model_urls:
+                exist_url.redirect = True
+
+    @api.multi
+    def _compute_url(self):
+
+        model_ref = "%s,%s" % (self._name, self.id)
+        _logger.info("used model  : %s ", model_ref)
+        # import pdb; pdb.set_trace()
+        url = self.env["url.url"].search([('model_id', '=', model_ref),
+                                          ('redirect', '=', False)])
+        if url:
+            self.url_key = url[0].url_key
+
+    @api.multi
+    def _compute_redirect_url(self):
+        """
+        :return:
+
+        """
+        model_ref = "%s,%s" % (self._name, self.id)
+
+        _logger.info("reference model : %s ", model_ref)
+
+        self.redirect_url_key_ids = self.env["url.url"]\
+            .search([('model_id', '=', model_ref), ('redirect', '=', True)])
+
+    @api.onchange('name')
+    def on_name_change(self):
+
+        for record in self:
+            name = record.name
+            url_key = record._prepare_url(name)
+            record.url_key = url_key
+            # _logger.info("Output..: %s ", url_key )
+
+    @api.onchange('url_key')
+    def on_url_key_change(self):
+        url = self._prepare_url(self.url_key)
+        if url != self.url_key:
+            self.url_key = url
+            return {'value': {},
+                    'warning': {
+                        'title': 'Adapt text rules',
+                        'message': 'it will will be adapted to %s' % (url)}}
```

Comment:
no it is hope that you write tùttù and it was change so we choose to prompt the user about the change ..